docs/proposals/job.md

```diff
@@ -0,0 +1,145 @@
+Job controller is similar to replication controller in that they manage pods.
+This implies they will follow the same controller framework that replication
+controllers already defined.  The biggest difference between a `Job` and a
+`ReplicationController` object is the purpose; `ReplicationController`
+ensures that a specified number of Pods are running at any one time, whereas
+`Job` is responsible for keeping the desired number of Pods to a completion of
+a task. This will be represented by the `RestartPolicy` which is required to
+always take value of `RestartPolicyNever` or `RestartOnFailure`.
+
+
+The new `Job` object will have the following content:
+
+```go
+// Job represents the configuration of a single job.
+type Job struct {
+    TypeMeta
+    ObjectMeta
+
+    // Spec is a structure defining the expected behavior of a job.
+    Spec JobSpec
+
+    // Status is a structure describing current status of a job.
+    Status JobStatus
+}
+
+// JobList is a collection of jobs.
+type JobList struct {
+    TypeMeta
+    ListMeta
+
+    Items []Job
+}
+```
+
+`JobSpec` structure is defined to contain all the information how the actual job execution
+will look like.
+
+```go
+// JobSpec describes how the job execution will look like.
+type JobSpec struct {
+
+    // TaskCount specifies the desired number of pods the job should be run with.
+    TaskCount int
+
+    // Optional duration in seconds relative to the StartTime that the job may be active
+    // before the system actively tries to terminate it; value must be positive integer
+    ActiveDeadlineSeconds *int64
+
+    // Selector is a label query over pods that should match the pod count.
+    Selector map[string]string
+
+    // Spec is the object that describes the pod that will be created when
+    // executing a job.
+    Spec PodSpec
+}
+```
+
+`JobStatus` structure is defined to contain informations about pods executing
+specified job.  The structure holds information about pods currently executing
+the job.
+
+```go
+// JobStatus represents the current state of a Job.
+type JobStatus struct {
+    // Executions holds a detailed information about each of the pods running a job.
+    Executions []JobExec
+
+    // Completions is the number of pods successfully completed their job.
+    Completions int
+
+}
+
+// JobExec represents the current state of a single execution of a Job.
+type JobExec struct {
+    // CreationTime represents time when the job execution was created
+    CreationTime util.Time
+
+    // StartTime represents time when the job execution was started
+    StartTime util.Time
+
+    // CompletionTime represents time when the job execution was completed
+    CompletionTime util.Time
+
+    // Phase represents the point in the job execution lifecycle.
+    Phase JobExecPhase
+
+    // Tag is added in labels of pod(s) created for this job execution.  It allows
+    // job object to safely group/track all pods started for one given job execution.
+    Tag util.UID
```

Comment:
Actually the above is to be applied to ScheduledJob, as I said before, there will be cases with more than one running Job in parallel, the reason for it being some kind of overlap, etc. With this Tag I was hoping to group them, so that each execution can be seen as a separate Job, but I think I'll go with an option of creating a separate Job for each of them, similarly like it's done with OpenShift's deployments, where the increasing number is responsible for tracking separate executions.